src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_CollectionExpression.cs

```diff
@@ -110,7 +110,8 @@ private BoundExpression VisitArrayOrSpanCollectionExpression(BoundCollectionExpr
                     // Assert that binding layer agrees with lowering layer about whether this collection-expr will allocate.
                     Debug.Assert(!IsAllocatingRefStructCollectionExpression(node, collectionTypeKind, elementType.Type, _compilation));
                     var constructor = ((MethodSymbol)_factory.WellKnownMember(WellKnownMember.System_ReadOnlySpan_T__ctor_Array)).AsMember(spanType);
-                    return _factory.New(constructor, _factory.Array(elementType.Type, elements));
+                    var rewrittenElements = elements.SelectAsArray(static (element, rewriter) => rewriter.VisitExpression((BoundExpression)element), this);
+                    return _factory.New(constructor, _factory.Array(elementType.Type, rewrittenElements));
```

Comment:
Elements were not being rewritten previously in this case, although this is specific to a few primitive numeric types where all elements are constant values, so rewriting may not be observable.